tests/ResourceSyncingTest.php

```diff
@@ -848,7 +849,7 @@ public function getSyncedAttributeNames(): array
     }
 }
 
-class ResourceUser extends Model implements Syncable
+class TenantUser extends Model implements Syncable
```

Comment:
![CleanShot 2022-11-04 at 1 23 47@2x](https://user-images.githubusercontent.com/33033094/199972022-05faa323-e21e-4562-af71-3b4c2eaa3554.png)

Reply:
I read that. But I though renaming it to `TenantUser` making it more clear. Should I revert it?

Reply:
Hmm, it depends. `TenantUser` is the most accurate naming, but the most problematic one since it has the highest chance of having conflicts. Seems like we're not using the class anywhere else though. So I guess we can keep it `TenantUser` for now and only deal with the "scoped" naming later if needed.

Reply:
Actually, `TenantUser` sounds like a model for `tenant_users`, i.e. the pivot table. So maybe we want to avoid the name in either case.

So I guess `ResourceUser` with a `// Tenant users` comment works best.

Reply:
Abrar, why are you renaming the other classes as well? Can you please carefully read the screenshot I linked?

https://github.com/archtechx/tenancy/pull/992/commits/f9ba778e1b6da7f19a191e68b07248b4dee1ddc3#diff-0c32d140e95604d5ab8852beb5eba2255180184351adfaa0ff7b2895012d4d7dR889-R918

I said that all classes named `ResourceUserProviding...` should be `TenantUserProviding...`. And in the first highlighted paragraph I said that **only for the *one* case of `ResourceUser` we may consider a different approach**.

It seems like you basically reverted the entire PR in the last commit.

Reply:
>Abrar, why are you renaming the other classes as well? Can you please carefully read the screenshot I linked?

I see. It's my fault. I used PHPStorm for refactoring, which did this. I ran tests and pushed the code. I thought PhpStorm renamed the class and its usage. Looked like it does rename everything starting with this name. Weird.

Reply:
You can just always check the diff before pushing changes. It's a good practice in general since it shows if any unexpected files were created/changed, or if you have some debugging changes that you don't want to push.